app/src/main/java/com/example/instasnap/Register.java

```diff
@@ -0,0 +1,164 @@
+package com.example.instasnap;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.text.TextUtils;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseAuthInvalidCredentialsException;
+import com.google.firebase.auth.FirebaseAuthUserCollisionException;
+import com.google.firebase.auth.FirebaseAuthWeakPasswordException;
+import com.google.firebase.auth.FirebaseUser;
+
+public class Register extends AppCompatActivity {
+
+    private EditText _editTextEmail;
+    private EditText _editTextPassword;
+    private Button _registerButton;
+    private FirebaseAuth _mAuth;
+    private ProgressBar _registerProgressBar;
+    private TextView _loginHereTextView;
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_register);
+
+        initialize();
+
+        setLoginNowButtonListener();
+
+        setRegisterButtonListener();
+
+    }
+
+    private void initialize() {
+        // Initializing registration objects
+        _mAuth = FirebaseAuth.getInstance();
+        _editTextEmail = findViewById(R.id.register_email_text_input);
+        _editTextPassword = findViewById(R.id.register_password_text_input);
+        _registerButton = findViewById(R.id.register_button);
+        _registerProgressBar = findViewById(R.id.register_progressBar);
+        _loginHereTextView = findViewById(R.id.login_now);
+    }
+
+    @Override
+    public void onStart() {
```

Comment:
Is it possible to be in Register activity when user is not null ?

Reply:
No you cant, because once the app starts it checks if there is a logged in user (not null), then a HomePage intent is being made insteqd of displaying login screen

Reply:
so the if statement is redundant (lines 62 to 66 can not be reached, right?)

Reply:
the if statement indicates whether there is a logged in user to the app on a certain smartphone, if there is a user logged in (hence why doing the currentUser != null) he will be navigated to the HomePage